odl/space/npy_ntuples.py

```diff
@@ -42,9 +43,225 @@
+            Use this weighting as-is. Compatibility with this
+            space's elements is not checked during init.
+
+            float: Weighting by a constant
+
+            array-like: Weighting by a matrix (2-dim. array) or a vector
+            (1-dim. array, corresponds to a diagonal matrix). A matrix
+            can also be given as a sparse matrix
+            ( `scipy.sparse.spmatrix`).
+
+            Default: no weighting
+
+            This option cannot be combined with ``dist``,
+            ``norm`` or ``inner``.
+
+        exponent : positive float, optional
+            Exponent of the norm. For values other than 2.0, no
+            inner product is defined.
+            If ``weight`` is a sparse matrix, only 1.0, 2.0 and
+            ``inf`` are allowed.
+
+            This option is ignored if ``dist``, ``norm`` or
+            ``inner`` is given.
+
+            Default: 2.0
+
+        Other Parameters
+        ----------------
+
+        dist : callable, optional
+            The distance function defining a metric on the space.
+            It must accept two `NumpyFnVector` arguments and
+            fulfill the following mathematical conditions for any
+            three vectors ``x, y, z``:
+
+            - ``dist(x, y) >= 0``
+            - ``dist(x, y) = 0``  if and only if  ``x = y``
+            - ``dist(x, y) = dist(y, x)``
+            - ``dist(x, y) <= dist(x, z) + dist(z, y)``
+
+            By default, ``dist(x, y)`` is calculated as ``norm(x - y)``.
+            This creates an intermediate array ``x - y``, which can be
+            avoided by choosing ``dist_using_inner=True``.
+
+            This option cannot be combined with ``weighting``,
+            ``norm`` or ``inner``.
+
+        norm : callable, optional
+            The norm implementation. It must accept an
+            `NumpyFnVector` argument, return a float and satisfy the
+            following conditions for all vectors ``x, y`` and scalars
+            ``s``:
+
+            - ``||x|| >= 0``
+            - ``||x|| = 0``  if and only if  ``x = 0``
+            - ``||s * x|| = |s| * ||x||``
+            - ``||x + y|| <= ||x|| + ||y||``
+
+            By default, ``norm(x)`` is calculated as ``inner(x, x)``.
+
+            This option cannot be combined with ``weighting``,
+            ``dist`` or ``inner``.
+
+        inner : callable, optional
+            The inner product implementation. It must accept two
+            `NumpyFnVector` arguments, return a element from
+            the field of the space (real or complex number) and
+            satisfy the following conditions for all vectors
+            ``x, y, z`` and scalars ``s``:
+
+            - ``<x, y> = conj(<y, x>)``
+            - ``<s*x + y, z> = s * <x, z> + <y, z>``
+            - ``<x, x> = 0``  if and only if  ``x = 0``
+
+            This option cannot be combined with ``weighting``,
+            ``dist`` or ``norm``.
+
+        dist_using_inner : bool, optional
```

Comment:
Wasn't this scrapped?

Reply:
Not sure whether or not on master yet, will check. If not, I'll make another PR just with that.

Reply:
Wasn't here yet, I'll do that next.